btf_interp/src/lib.rs

```diff
@@ -16,42 +17,42 @@ pub struct VirtualMachine<T> {
 
 impl<T> VirtualMachine<T> {
     /// Create VM based on the size, by default is 30 000. Also, it can be adjusted, by default it doesn't.
-    pub fn new(size: Option<usize>, adjust_tape: Option<bool>) -> Self {
+    pub fn new(size: Option<NonZeroUsize>, adjust_tape: Option<bool>) -> Self {
         VirtualMachine {
             tape: Vec::new(),
-            tape_size: size.unwrap_or(30000),
+            tape_size: size.unwrap_or(NonZeroUsize::new(3000).unwrap()),
             adjust_tape: adjust_tape.unwrap_or(false),
             pointer: 0,
         }
     }
 
     /// Get the tape.
-    pub fn tape(&self) -> &[T] {
+    fn tape(&self) -> &[T] {
         &self.tape[..]
     }
 
     /// Get the size of the tape.
-    pub fn tape_size(&self) -> usize {
-        self.tape_size
+    fn tape_size(&self) -> usize {
+        self.tape_size.get()
     }
 
     /// Get the indicator whether it is possible to adjust tape or not.
-    pub fn adjust_tape(&self) -> bool {
+    fn adjust_tape(&self) -> bool {
         self.adjust_tape
     }
 
     /// Get pointer of the tape.
-    pub fn pointer(&self) -> usize {
+    fn pointer(&self) -> usize {
         self.pointer
     }
 
     /// Add element into tape. If tape size exceeded, element is not going to be added, error message is shown.
     pub fn push(&mut self, item: T) -> Result<(), String> {
-        if self.tape.len() >= self.tape_size {
+        if self.tape.len() >= self.tape_size() {
             if !self.adjust_tape {
                 return Err(format!("Max tape size of {} reached.", self.tape_size));
             }
-            self.tape_size += 1;
+            self.tape_size.checked_add(1);
```

Comment:
If you are going to use a checked_add() then I'd suggest that you should verify the return value.  however in this instance you're better off just adding one the original way.  You won't overflow anyway since it'd take 18.5 exabytes just for the tape.